lib/modules/platform/github/index.ts

```diff
@@ -204,7 +204,17 @@ export async function initPlatform({
     renovateUsername,
     token,
   };
-
+  if (platformResult.endpoint === 'https://api.github.com/') {
+    logger.debug('Adding GitHub token as GHCR password');
+    platformResult.hostRules = [
```

Comment:
Hello,

It seems that after upgrading to this version, we have a broken build now. We are specifying this and credentials explicitly and it seems that now it is ignored and this one is being used. Renovate cannot make the assumption that the GitHub Action token should be used, because that one only has access to the packages in the same repository. We need to provide our own token with permissions across all repositories.

Please advise.

Reply:
Thank you!

Reply:
Sorry for the interruption!

Reply:
@rarkins we are still having the problem. It is no longer affecting the NuGet packages, but the ghcr.io Docker images still don't use the explicitly specified credentials:

![image](https://github.com/renovatebot/renovate/assets/176021/ca39c99d-7140-4877-90a0-439a377a81ae)

Reply:
Please start a new discussion for that

Reply:
Created it here: https://github.com/renovatebot/renovate/discussions/25134. Pretty sure it is related to this change though, it seems that this change overrides any existing explicit hostRule and uses the token that Renovate assumes is correct.